template/cordova/lib/deployment.js

```diff
@@ -0,0 +1,275 @@
+/*
+       Licensed to the Apache Software Foundation (ASF) under one
+       or more contributor license agreements.  See the NOTICE file
+       distributed with this work for additional information
+       regarding copyright ownership.  The ASF licenses this file
+       to you under the Apache License, Version 2.0 (the
+       "License"); you may not use this file except in compliance
+       with the License.  You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+       Unless required by applicable law or agreed to in writing,
+       software distributed under the License is distributed on an
+       "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+       KIND, either express or implied.  See the License for the
+       specific language governing permissions and limitations
+       under the License.
+*/
+
+var Q     = require('q'),
+    fs    = require('fs'),
+    /* jshint ignore:start */ // 'path' only used in ignored blocks
+    path  = require('path'),
+    /* jshint ignore:end */
+    proc  = require('child_process');
+
+// neither 'exec' nor 'spawn' was sufficient because we need to pass arguments via spawn
+// but also need to be able to capture stdout / stderr
+function run(cmd, args, opt_cwd) {
+    var d = Q.defer();
+    try {
+        var child = proc.spawn(cmd, args, {cwd: opt_cwd, maxBuffer: 1024000});
+        var stdout = '', stderr = '';
+        child.stdout.on('data', function(s) { stdout += s; });
+        child.stderr.on('data', function(s) { stderr += s; });
+        child.on('exit', function(code) {
+            if (code) {
+                d.reject(stderr);
+            } else {
+                d.resolve(stdout);
+            }
+        });
+    } catch(e) {
+        console.error('error caught: ' + e);
```

Comment:
This function is based on spawn.js, modified only to capture stdout/stderr.  The comment is a fair question, but I thought I'd preserve consistency in this case.